File: GrayCode/Java/GrayCode.java
public class GrayCode {
    public static long grayEncode(long n) {
        return n ^ (n >>> 1);
    }

    public static long grayDecode(long n) {
        long p = n;
        while ((n >>>= 1) != 0)
            p ^= n;
        return p;
    }

    public static void main(String[] args) {
        long enc;
        for (int i = -1; i < java.lang.Math.pow(2, 25); i++) {
            enc = grayEncode(i);
            grayDecode(enc);
        }
    }
}

File: GrayCode/Go/GrayCode.go
package main

import "math"

func enc(b int) int {
    return b ^ b>>1
}

func dec(g int) (b int) {
    for ; g != 0; g >>= 1 {
        b ^= g
    }
    return
}

func main() {
	it := int(math.Pow(2, 20))
    for b := 0; b < it; b++ {
        g := enc(b)
        dec(g)
    }
}
